### DISCUSSION TITLE
Calibration question: different for low and high Bg values

### DISCUSSION THREAD

**Discussion by blaqone** [NightscoutFoundation/xDrip#2826] [Q&A]
Hi,

my sensor reads normal values (80-120 mg/dl) too low (-10-20mg/dl) and high values (> 200mg/dl) too high. 
is there a way to calibrate this scenario?

**Comment by PhilTU** [comment]
The straight line equation method will bring both of them closer, for example a slope of 0.625 and an intercept of 50,

Which sensor type ? What happens when you calibrate now ?

**Reply by blaqone** [reply]
No this doesn’t work. I can set the slope, but once it set a intercept I get „???“ and no more values. It even resets slope and intercept to 1 - 1. no idea why.

**Reply by Navid200** [reply]
What do you mean by you can set the slope?
You can only do that in engineering mode for troubleshooting and testing.  You should not force slope or intercept.  You should calibrate and allow the calibration mechanism calculate those using your entered calibrations.

**Reply by blaqone** [reply]
Yes I am running in developer mode and I am using plug-in mode without a plug-in to disable the integrated smoothing. AAPS does the smoothing for me.

**Reply by Navid200** [reply]
Engineering mode is for testing.  It's not meant to be used for regular operation.  It is highly likely that it will fail.  That's why something is placed in engineering mode.

**Reply by blaqone** [reply]
It works perfectly fine!

**Reply by blaqone** [reply]
Oh and this is not related to the actual question. We can move the discussion to a new thread if you want. Can you open it?

**Reply by PhilTU** [reply]
> No this doesn’t work. I can set the slope, but once it set a intercept I get „???“ and no more values. It even resets slope and intercept to 1 - 1. no idea why.

You didn't answer the question about the sensor type, but in some cases the slope is fixed at 1 and xDrip+ determines the intercept by straight line regression.

You can change a setting to allow slopes other than 1, search Settings for "slope" and it'll find "Non-fixed Libre slopes" for you

Rather than dialling in values you could just enter blood glucose values and let it do its thing. 

A straight line correction will improve both ends in the scenario you describe. Non linear alternatives might do better, but they're above my pay grade.

**Reply by blaqone** [reply]
Okay thank you - I will milling to it and also have a look at non-mi ear calibration methods. 
I’m using raw L3 minute values which I calibrate and send to AAPS for smoothing.
